packages/plugin-mcp/src/mcp/tools/job/create.ts

```diff
@@ -0,0 +1,414 @@
+    const tasksDir = join(jobsDir, 'tasks')
+    const workflowsDir = join(jobsDir, 'workflows')
+
+    if (!existsSync(tasksDir)) {
+      mkdirSync(tasksDir, { recursive: true })
+    }
+    if (!existsSync(workflowsDir)) {
+      mkdirSync(workflowsDir, { recursive: true })
+    }
+
+    const camelCaseJobSlug = toCamelCase(jobSlug)
+    const targetDir = jobType === 'task' ? tasksDir : workflowsDir
+    const fileName = `${camelCaseJobSlug}.ts`
+    const filePath = join(targetDir, fileName)
+
+    if (verboseLogs) {
+      payload.logger.info(`[payload-mcp] Target file path: ${filePath}`)
+    }
+
+    // Security check: ensure we're working with the jobs directory
+    if (!filePath.startsWith(jobsDir)) {
+      payload.logger.error(`[payload-mcp] Invalid job path attempted: ${filePath}`)
+      return {
+        content: [
+          {
+            type: 'text' as const,
+            text: '❌ **Error**: Invalid job path',
+          },
+        ],
+      }
+    }
+
+    // Check if file already exists
+    if (existsSync(filePath)) {
+      if (verboseLogs) {
+        payload.logger.info(`[payload-mcp] Job file already exists: ${fileName}`)
+      }
+      return {
+        content: [
+          {
+            type: 'text' as const,
+            text: `❌ **Error**: Job file already exists: ${fileName}`,
+          },
+        ],
+      }
+    }
+
+    // Generate job content based on type
+    let jobContent: string
+    if (jobType === 'task') {
+      jobContent = generateTaskContent(
+        jobName,
+        jobSlug,
+        description,
+        inputSchema,
+        outputSchema,
+        jobData,
+      )
+    } else {
+      jobContent = generateWorkflowContent(
+        jobName,
+        jobSlug,
+        description,
+        inputSchema,
+        outputSchema,
+        jobData,
+      )
+    }
+
+    // Write the job file
+    writeFileSync(filePath, jobContent, 'utf8')
+    if (verboseLogs) {
+      payload.logger.info(`[payload-mcp] Successfully created job file: ${filePath}`)
+    }
+
+    // Update the main job file
+    createOrUpdateJobFile(req, verboseLogs, jobsDir, jobName, jobType, jobSlug, camelCaseJobSlug)
+
+    // Validate the generated file
+    const validationResult = await validatePayloadFile(fileName, jobType)
+    if (validationResult.error) {
+      return {
+        content: [
+          {
+            type: 'text' as const,
+            text: `❌ **Error**: Generated job has validation issues:\n\n${validationResult.error}`,
+          },
+        ],
+      }
+    }
+
+    return {
+      content: [
+        {
+          type: 'text' as const,
+          text: `✅ **Job created successfully!**\n\n**File**: \`${fileName}\`\n**Type**: \`${jobType}\`\n**Slug**: \`${jobSlug}\`\n**Description**: ${description}\n\n**Next steps**:\n1. Restart your development server to load the new job\n2. The job will be available in your Payload jobs configuration\n3. You can now queue and run this job`,
```

Comment:
This is never meant to be used in production, right?
If so how is this going to work? Imagine if the code is changing but then the server restarts and then it seems these changes would be lost. Is that accurate?